src/monkeyflocker/client.py

```diff
@@ -3,15 +3,36 @@
 """
 
 import asyncio
+import logging
 import math
+import os
+import pathlib
+import sys
 from dataclasses import InitVar, dataclass, field
-from typing import List
+from typing import Any, Dict, List, Optional
 
 import aiohttp
 import jinja2
+import structlog
+from structlog._config import BoundLoggerLazyProxy
 
 from .user import MonkeyflockerUser as MFUser
 
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+@dataclass
+class ShadowRequest:
+    """This contains method, url, username, and body, if any.
+    Retrieving it from the actual request turns out to be tricky because
+    it's really a morass of little internal _*RequestContextManager objects.
```

Comment:
I don't understand from this documentation what a "shadow request" is.

Reply:
Resolved by getting rid of this.  The gather() only really helps us on the DELETE.